cs301-itsa-ay2324/admin-ui-proxy
# feature: User Enrolment Integration (w/o auth)

## Overview
This PR addresses the UI part of frontend-backend integration for user service. For more project details please refer to Issue 28 [here](https://github.com/cs301-itsa/project-2023-24t1-g1-t3/issues/28#issue-1945411369).

## Features
- [x] C - Enrol new user
- [x] R - View all users
- [x] U - Update user details
- [x] D - Delete a specific user
- [x] Alert dialog for delete button
- [x] Toasters for action completion (create & update)

## To be addressed in future PR
- [ ] User's points balance 

## Testing
To carry out testing for this PR, please follow the steps below:

1. Ensure backend services (Docker) are up and running
2. Ensure `role` table is populated with `OWNER`, `MANAGER`, `ENGINEER`, and `PRODUCT_MANAGER`
3. Set up the project
```
yarn install
yarn dev
```

## Demo

https://github.com/cs301-itsa-ay2324/admin-ui-proxy/assets/68802506/9149ac22-1afb-4a89-9098-0b50a7e073e0
